akka-projection-grpc/src/main/scala/akka/projection/grpc/consumer/scaladsl/GrpcReadJournal.scala

```diff
@@ -186,10 +187,19 @@ final class GrpcReadJournal private (
   def this(system: ExtendedActorSystem, config: Config, cfgPath: String) =
     this(system, config, cfgPath, ProtoAnySerialization.Prefer.Scala)
 
+  /**
+   * Correlation id to be used with [[ConsumerFilter.ReplayWithFilter]].
+   * Such replay request will trigger replay in all `eventsBySlices` queries
+   * with the same `streamId` running from this instance of the `GrpcReadJournal`.
+   * Create separate instances of the `GrpcReadJournal` to have separation between
+   * replay requests for the same `streamId`.
+   */
+  val replayCorrelationId: UUID = UUID.randomUUID() // FIXME add to javadsl
```

Comment:
@johanandren @pvlugter Before completing this I'd like to hear if you have any better suggestion than having this correlation id in the GrpcReadJournal instance, which is shared between the triggerReplay and the eventsBySlices.

The problem could be if the same GrpcReadJournal isntance is used for unrelated projections.

Ideally the eventsBySlices would materialize this correlation id and then that should used in the triggerReplay, but given all levels of abstraction and public api I don't see that we could introduce that.

Reply:
It's also an anti-pattern to use several projections with the same streamId to the same producer host, since there is much overhead of transferring the same events multiple times. If different producer hosts it will be different GrpcReadJournal instances anyway.

Reply:
I couldn't see a better way to do this, when uncovering the issue yesterday. Ideally they're connected, but likely need to change things a bit to add another layer of abstraction, so that it can provide a new `EventsBySliceQuery with CanTriggerReplay` particular to each query?

The UUID for each GrpcReadJournal should work for the use case that we've seen, which is implementing the anti-pattern of multiple projections for the same stream id and producer. Where the GrpcReadJournal is shared across slice ranges (already checked for Replays) but separate across projections for the same stream id.

Reply:
The same read journal instance re-used would only be a problem if used for unrelated projections from the same stream id, because if it is different streams that is fine. I think this is good enough and can't think of a better solution without needing to mess around quite a bit with the read journal APIs.